flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/CoGroupITCase.java

```diff
@@ -797,4 +873,46 @@ public void coGroup(Iterable<Integer> first, Iterable<Tuple3<Integer, Long, Stri
 			}
 		}
 	}
+
+	public static class TestDistribution implements DataDistribution {
+		public Integer boundaries[][] = new Integer[][]{
+				new Integer[]{2, 2},
+				new Integer[]{5, 4},
+				new Integer[]{10, 12},
+				new Integer[]{21, 6}
+		};
+
+		public TestDistribution() {}
+
+		@Override
+		public Object[] getBucketBoundary(int bucketNum, int totalNumBuckets) {
+			return boundaries[bucketNum];
+		}
+
+		@Override
+		public int getNumberOfFields() {
+			return 2;
+		}
+
+		@Override
+		public TypeInformation[] getKeyTypes() {
+			return new TypeInformation[]{BasicTypeInfo.INT_TYPE_INFO, BasicTypeInfo.INT_TYPE_INFO};
+		}
+
+		@Override
+		public void write(DataOutputView out) throws IOException {
+
+		}
+
+		@Override
+		public void read(DataInputView in) throws IOException {
+
+		}
+
+		@Override
+		public boolean equals(Object obj) {
+			// The test is running with same distribution, so return true directly
+			return true; 
```

Comment:
That is true. 
Could you change it to `obj instanceof TestDistribution` nonetheless? Sometimes test code is copied and this could be a bit dangerous ;-)